### PR TITLE
Set headers on stream proxy

### LINKED ISSUE #401: Proxy headers in got.stream
I use `got@6.7.1`. I need to proxy an url from some upstream. So, I do
```
got.stream(url).pipe(res);
```
where `res` is an express response.

But it proxies only body, not headers. To proxy headers as well I have to add an extra handler
```
got.stream(url)
    .on('response', (msg) => {
        res.set(msg.headers);
    })
    .pipe(res);
```

Is it expected behaviour? For example `request` library proxies body and headers out of the box.

### PR DESCRIPTION
**Pull request by szmarczak**
Closes #401.

### CODE REVIEW

File: source/as-stream.js [hunk 1]
```diff
@@ -50,13 +52,27 @@ module.exports = options => {
...
 			proxy.emit('error', new HTTPError(statusCode, response.statusMessage, response.headers, options), null, response);
 			return;
 		}
 
+		finished = true;
+
+		response.pipe(output);
+
+		for (const destination of piped) {
+			if (!destination.headersSent) {
```
Comment (sindresorhus) [inline source/as-stream.js]: Usually better with early returns to reduce nesting.

```js
if (!destination.headersSent) {
  continue;
}
```

File: source/as-stream.js [hunk 2]
```diff
@@ -50,13 +52,27 @@ module.exports = options => {
...
+				for (const [key, value] of Object.entries(response.headers)) {
+					if (key.toLowerCase() !== 'content-encoding') {
```
Comment (sindresorhus) [inline source/as-stream.js]: Can you add a short code comment why we don't set this? Are there other headers we shouldn't overwrite?

File: source/as-stream.js [hunk 3]
```diff
@@ -9,6 +9,8 @@ module.exports = options => {
 	const input = new PassThrough();
 	const output = new PassThrough();
 	const proxy = duplexer3(input, output);
+	const piped = new Set();
+	let finished = false;
```
Comment (sindresorhus) [inline source/as-stream.js]: `isFinished`

File: source/as-stream.js (hunk 2 above)
Comment (szmarczak) [reply source/as-stream.js]: > Can you add a short code comment why we don't set this?

Yes.

> Are there other headers we shouldn't overwrite?

No.

File: source/as-stream.js (hunk 1 above)
Comment (szmarczak) [reply source/as-stream.js]: You misled me :P You shouldn't negate the condition :D

**Pull request merged**